internal/resource/common/pluginconfiguration/plugin_configuration_state.go

```diff
@@ -0,0 +1,203 @@
+package pluginconfiguration
+
+import (
+	"fmt"
+
+	"github.com/hashicorp/terraform-plugin-framework/attr"
+	"github.com/hashicorp/terraform-plugin-framework/diag"
+	"github.com/hashicorp/terraform-plugin-framework/types"
+	"github.com/hashicorp/terraform-plugin-framework/types/basetypes"
+	client "github.com/pingidentity/pingfederate-go-client/v1125/configurationapi"
+	internaltypes "github.com/pingidentity/terraform-provider-pingfederate/internal/types"
+)
+
+var (
+	diags = diag.Diagnostics{}
```

Comment:
We don't want a global diags like this, or it will just keep getting added to over time from separate requests. We should be creating a new diags in the methods when we need it